File: src/backend/go/internal/dto/todo_item_attachment.go
package dto

import "time"

type SaveTodoItemAttachmentRequest struct {
	FileID uint `json:"fileId" binding:"required"`
}

type TodoItemAttachmentResponse struct {
	ID              uint       `json:"id"`
	TodoItemID      uint       `json:"todoItemId"`
	FileID          uint       `json:"fileId"`
	CreatedAt       time.Time  `json:"createdAt"`
	CreatedByUserID *uint      `json:"createdByUserId"`
	UpdatedAt       *time.Time `json:"updatedAt"`
	UpdatedByUserID *uint      `json:"updatedByUserId"`
}

File: src/backend/python/api/schemas/todo_item_attachment.py
from datetime import datetime

from pydantic import BaseModel


class CreateTodoItemAttachmentRequest(BaseModel):
    file_id: int


class TodoItemAttachmentResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: int
    todo_item_id: int
    file_id: int
    created_at: datetime
    created_by_user_id: int | None = None
    updated_at: datetime | None
    updated_by_user_id: int | None = None

